raiden_contracts/tests/test_channel_update_transfer.py

```diff
@@ -154,6 +154,95 @@ def test_update_notclosed_fail(
         )
 
 
+def test_update_wrong_nonce_fail(
+        token_network,
+        create_channel,
+        channel_deposit,
+        get_accounts,
+        get_block,
+        create_balance_proof,
+        create_balance_proof_update_signature
+):
+    (A, B, Delegate) = get_accounts(3)
+    settle_timeout = 6
+    deposit_A = 20
+    channel_identifier = create_channel(A, B, settle_timeout)[0]
+    channel_deposit(A, deposit_A, B)
+    balance_proof_A = create_balance_proof(channel_identifier, A, 10, 0, 5, fake_bytes(32, '02'))
+    balance_proof_B = create_balance_proof(channel_identifier, B, 5, 0, 3, fake_bytes(32, '02'))
+    balance_proof_update_signature_B = create_balance_proof_update_signature(
+        B,
+        channel_identifier,
+        *balance_proof_A
+    )
+    txn_hash1 = token_network.transact({'from': A}).closeChannel(B, *balance_proof_B)
+
+    token_network.transact({'from': Delegate}).updateNonClosingBalanceProof(
+        A,
+        B,
+        *balance_proof_A,
+        balance_proof_update_signature_B
+    )
+
+    balance_proof_A_same_nonce = balance_proof_A
+    with pytest.raises(TransactionFailed):
+        token_network.transact({'from': Delegate}).updateNonClosingBalanceProof(
+            A,
+            B,
+            *balance_proof_A_same_nonce,
+            balance_proof_update_signature_B
+        )
+
+    balance_proof_A_lower_nonce = create_balance_proof(
+        channel_identifier,
+        A,
+        10,
+        0,
+        4,
+        fake_bytes(32, '02')
+    )
+
+    balance_proof_update_signature_B = create_balance_proof_update_signature(
+        B,
+        channel_identifier,
+        *balance_proof_A_lower_nonce
+    )
+    with pytest.raises(TransactionFailed):
+        token_network.transact({'from': A}).updateNonClosingBalanceProof(
+            A,
+            B,
+            *balance_proof_A_lower_nonce,
+            balance_proof_update_signature_B
+        )
+
+    (_, settle_block_number, state) = token_network.call().getChannelInfo(A, B)
+
+    assert settle_block_number == settle_timeout + get_block(txn_hash1)  # settle_block_number
+    assert state == CHANNEL_STATE_CLOSED  # state
+
+    (
+        _,
+        _,
+        A_is_the_closer,
+        A_balance_hash,
+        A_nonce
+    ) = token_network.call().getChannelParticipantInfo(A, B)
+    assert A_is_the_closer is True
+    assert A_balance_hash == balance_proof_A[0]
+    assert A_nonce == 5
+
+    (
+        _,
+        _,
+        B_is_the_closer,
+        B_balance_hash,
+        B_nonce
+    ) = token_network.call().getChannelParticipantInfo(B, A)
+    assert B_is_the_closer is False
+    assert B_balance_hash == balance_proof_B[0]
+    assert B_nonce == 3
```

Comment:
You can also make an `updateBalanceProof_state_tests` in `fixtures.channels.py` (check `settle_state_tests`) or another file to use for all the tests.